test/unit/test_events.py

```diff
@@ -45,3 +46,32 @@ def test_formatting(self):
         # were no args passed after the initial message
         event = AdapterEventDebug(name="dbt_tests", base_msg="boop{x}boop", args=())
         self.assertTrue("boop{x}boop" in event.message())
+
+class TestEventCodes(TestCase):
+
+    def get_all_subclasses(self,cls):
+        all_subclasses = []
+        for subclass in cls.__subclasses__():
+            all_subclasses.append(subclass)
+            all_subclasses.extend(self.get_all_subclasses(subclass))
+        return set(all_subclasses)
+
+
+    def setUp(self):
+        pass
+
+    # this interface is documented for adapter maintainers to plug into
+    # so we should test that it at the very least doesn't explode.
+    def test_event_codes(self):
+        # all_concrete = # all subclasses (any depth) of Event
+        all_concrete = self.get_all_subclasses(Event)
+        all_codes = set()
+
+        for event in all_concrete:
+            if not inspect.isabstract(event):
+                # must be in the form 1 capital letter, 3 digits
+                self.assertTrue('^[A-Z][0-9]{3}', event.code)
+                # cannot have been used already
+                self.assertFalse(event.code in all_codes,f'{event.code}')
```

Comment:
minor formatting, and helper string improvement
```suggestion
                self.assertFalse(event.code in all_codes, f'Code {event.code} is assigned more than once. Check types.py for duplicates.')
```

Reply:
Thank you for all the catches, made changes and repushed.